### PR TITLE
error page layout fixups

### PR DESCRIPTION
**Pull request by spasovski**
I couldn't find any issues with this locally but the `#page` min-height obviously affects every page so we should be watchful. This is based on a vidyo chat I had with Phil where we wanted desktop pages to have a certain minimum height when little to no content was available.

sample before:

sample after:

### CODE REVIEW

**Comment by ngokevin** [comment]
Can we make the background white like all the other sections?

**Comment by spasovski** [comment]
re-r when you get a chance?

**Comment by ngokevin** [comment]
Feels like the Oh No should be in the subheader

File: src/templates/_macros/error_page.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% macro error_page(code=_('Error'),
```
Comment (ngokevin) [inline src/templates/_macros/error_page.html]: I would probably have the fallbacks in the macro 

```{{ code || _('Error') }} or {{ message || _('Message') }}```

File: src/templates/_macros/error_page.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                    message=_('The page you were looking for was not found.')) %}
```
Comment (ngokevin) [inline src/templates/_macros/error_page.html]: How about:

```Sorry! We could not find the page you were looking for.```

File: src/templates/_macros/error_page.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <section class="main">
+    <div class="subheader">
+      <h1>{{ code }}</h1>
+    </div>
+  </section>
+
+  <section class="main full prose">
+    <div class="detailed-error">
+      <h1>{{ _('Oh no!') }}</h1>
```
Comment (ngokevin) [inline src/templates/_macros/error_page.html]: We can get rid of the Oh No I guess now

File: src/templates/feed/collection.html [hunk 4]
```diff
@@ -6,8 +7,7 @@
 {% placeholder %}
   <p class="spinner alt spaced"></p>
 {% except %}
-  <div class="detailed-error">
-    <h2>{{ _('Oh no!') }}</h2>
-    <p>{{ _('We could not load that collection. Please try again later.') }}</p>
-  </div>
+  {{ error_page(
+      message=_('We could not load that collection. Please try again later.')
+  ) }}
```
Comment (ngokevin) [inline src/templates/feed/collection.html]: nit: this style indent looks a bit off

File: src/templates/errors/404.html [hunk 5]
```diff
@@ -1,12 +1,3 @@
-<section class="main">
-  <div class="subheader">
-    <h1>404</h1>
-  </div>
-</section>
+{% include "_macros/error_page.html" %}
```
Comment (ngokevin) [inline src/templates/errors/404.html]: use "from ... import ..." to be explicit

File: src/templates/_macros/error_page.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{% macro error_page(code=_('Error'),
+                    message=_('The page you were looking for was not found.')) %}
+  <section class="main">
+    <div class="subheader">
+      <h1>{{ code }}</h1>
```
Comment (ngokevin) [inline src/templates/_macros/error_page.html]: Can we add an error_messages.js module that maps status codes to messages, and then use that to display the appropriate message here based on the code?

**Comment by ngokevin** [comment]
nice, i like that we're consolidating the error pages into a single point

File: src/templates/_macros/error_page.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    </div>
+  </section>
+
+  <section class="main full prose">
+    <div class="detailed-error">
```
Comment (ngokevin) [inline src/templates/_macros/error_page.html]: class="error-page" would be cool

**Comment by ngokevin** [comment]
status?

**Comment by spasovski** [comment]
Updated.

File: src/media/js/error_messages.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+/*
+    Error codes and their associated messages.
+*/
+define('error_messages', ['core/l10n'], function(l10n) {
+    'use strict';
+    var gettext = l10n.gettext;
+
+    return {
```
Comment (ngokevin) [inline src/media/js/error_messages.js]: might be better for this to take generic status codes so we don't have to write a message for each error

Comment (spasovski) [inline src/media/js/error_messages.js]: I like the personal touch of specific error messages but would be also OK with 'item_missing': 'We had trouble loading the details of this item...'

Comment (ngokevin) [inline src/media/js/error_messages.js]: Yeah but we don't want special l10n for every page. It increases the file size for each locale and introduces more l10n work. If you want to make it specialized, pass a noun into gettext and format it in. But it'd be good for the keys to be 403/404/500, etc using response.status_code

**Pull request closed without merging**